### PR TITLE
[dstorage] update for version 1.2.0 release

### PR DESCRIPTION
**Pull request by walbourn**
### New Features
- Add support for enabling buffered file IO for use on HDDs that may benefit from OS file caching behaviors.
- Add IDStorageQueue2::GetCompressionSupport API to indicate what path the DirectStorage runtime will take when decompressing a supported GPU decompression format.
- Update dstorage.h and dstorageerr.h to be covered by the MIT License.
- Add Microsoft.Direct3D.DirectStorage.winmd, to ease generation of non-C++ bindings to the API.

### Bug Fixes
- Add "Reserved1" field to DSTORAGE_REQUEST_OPTIONS.  This makes the in-memory layout of the structure more explicit, but doesn't actually change the layout from previous versions.
- Fix DSTORAGE_REQUEST_DESTINATION_TEXTURE_REGION for 3D textures.
- Fix scheduling issue that manifested when transferring uncompressed data from memory to buffers

### Performance improvements
- Move the copy after GPU decompression onto the compute queue for GPUs where this is faster.

### CODE REVIEW

**Comment by walbourn** [comment]
Included code review changes proposed for **xaudio2redist** in [this PR](https://github.com/microsoft/vcpkg/pull/28699) so it will work in 'static' contexts as well.

**Comment by walbourn** [comment]
Validated all supported triplets:

```
vcpkg install dstorage:x86-windows
vcpkg install dstorage:x86-windows-static-md
vcpkg install dstorage:x64-windows
vcpkg install dstorage:x64-windows-static-md
vcpkg install dstorage:arm-windows
vcpkg install dstorage:arm64-windows
vcpkg install dstorage:arm64-windows-static-md

vcpkg install dstorage:x64-mingw-dynamic
vcpkg install dstorage:x86-mingw-dynamic
```

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/dstorage/portfile.cmake [hunk 1]
```diff
@@ -1,10 +1,10 @@
 vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
```
Comment (JonLiu1993) [inline ports/dstorage/portfile.cmake:1]:
```suggestion
```

Comment (JonLiu1993) [reply ports/dstorage/portfile.cmake:1]: @walbourn, thanks for your PR, this is no longer needed.

File: ports/dstorage/portfile.cmake [hunk 2]
```diff
@@ -1,10 +1,9 @@
-vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}
-vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
+set(VCPKG_POLICY_DLLS_IN_STATIC_LIBRARY enabled)
```
Comment (BillyONeal) [inline ports/dstorage/portfile.cmake:1]: This policy is kind of scary as you get CRT weirdness... can you describe why this was changed from the `vcpkg_check_linkage` call?

Comment (walbourn) [reply ports/dstorage/portfile.cmake:1]: The DirectStorage DLLs use pure API sets, so they use OS-provided DLLs only. There's no dependencies on any version of the MSVC Runtime either in the OS or retail.

The request was first made for **xaudio2redist** which has an OS DLL redist component as well. That DLL also uses only OS DLLs and makes no use of MSVC Runtimes.

**Review by BillyONeal — APPROVED** [review]

File: ports/dstorage/portfile.cmake (hunk 2 above)
Comment (BillyONeal) [reply ports/dstorage/portfile.cmake:3]: Thanks!

**Review by JonLiu1993 — APPROVED** [review]

**Review by coopp — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**